src/app/icd/client/RefreshKeySender.h

```diff
@@ -0,0 +1,137 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "ICDClientInfo.h"
+#include "ICDClientStorage.h"
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/CommandSender.h>
+#include <app/OperationalSessionSetup.h>
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPConfig.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/ScopedNodeId.h>
+#include <lib/support/CodeUtils.h>
+#include <stddef.h>
+
+namespace chip {
+namespace app {
+
+class CheckInDelegate;
+/**
```

Comment:
I had to move this out of RefreshKeySender in order to use in DefaultCheckInDelegate to create the RefreshKeyBuffer object to pass to the RefreshKeySender object.